spec/vehicle_factory_spec.rb

```diff
@@ -0,0 +1,29 @@
+require './lib/vehicle'
+require './lib/dmv_data_service'
+require './lib/vehicle_factory'
+
+RSpec.describe VehicleFactory do
+    before(:each) do
+        @factory = VehicleFactory.new
+    end
+
+    describe "initialize" do
+        it 'exists' do
+            expect(@factory).to be_a(VehicleFactory)
+        end
+    end
+
+    describe "create_vehicles" do
+        it "can create vehicles" do
+            wa_ev_registrations = DmvDataService.new.wa_ev_registrations
+            vehicles = @factory.create_vehicles(wa_ev_registrations)
+
+            expect(vehicles[0]).to be_a(Vehicle)
+            expect(vehicles[0].engine).to eq(:ev)
+            expect(vehicles[0].make).to eq("BMW")
+            expect(vehicles[0].model).to eq("i3")
+            expect(vehicles[0].vin).to eq("WBY8P8C51K")
+            expect(vehicles[0].year).to eq("2019")
```

Comment:
These are robust tests! The only risk here is that the facility data is coming from an external service (via the DmvDataService) so there's a slight chance that this data could change/update and cause your tests to fail even if there's nothing wrong with your code or the way you're handling their data. This is a risk that is totally fine for the project, but as you progress in Turing (especially Mod 3) you'll start to think about other options when you're testing something that uses a data set we can't control. Two other options are:

 1. using more general assertions (i.e. expect(facility_zero.name).to be_a String)
 2.  using mock (dummy) data instead of actually getting data from the DmvDataService

This is food for thought for as you progress through the program. For now, these are great tests!